src/misc/ListModel.ts

```diff
@@ -286,26 +289,33 @@ export class ListModel<ElementType extends ListElement> {
 			if (entity) {
 				selectedItems.delete(entity)
 
-				// select next item in the list
-				if (
-					this.getSelectedAsArray().length === 1 &&
-					this.getSelectedAsArray()[0] === entity &&
-					this.rawState.filteredItems.length > 1 &&
-					!this.rawState.inMultiselect
-				) {
-					newActiveElement =
-						entity === last(this.state.items)
-							? this.state.items[this.state.items.length - 2]
-							: this.state.items[this.state.items.indexOf(entity) + 1]
-
-					selectedItems.add(newActiveElement)
+				if (this.rawState.filteredItems.length > 1) {
+					const desiredBehavior = this.config.autoSelectBehavior?.() ?? null
+					if (desiredBehavior === ListAutoSelectBehavior.NONE || this.state.inMultiselect) {
+						selectedItems.clear()
```

Comment:
is this correct that we remove the whole selection here?

Reply:
we did it manually in a lot of cases so it might be a change of behavior, we should check this carefully